src/tests/test_util.py

```diff
@@ -143,3 +178,456 @@ def test_parse_sql_results_to_json(ip, capsys, rows, columns, expected_json):
+            "sqlplot",
+            ["--table", "--column", "--show-numbers", "-S"],
+            [("S", "show-numbers")],
+        ),
+        # for mixed
+        (
+            SqlPlotMagic.execute,
+            "sqlplot",
+            [
+                "--table",
+                "--column",
+                "--column",
+                "-w",
+                "--with",
+                "--show-numbers",
+                "--show-numbers",
+                "--binwidth",
+                "--orient",
+                "-o",
+                "--breaks",
+                "-B",
+            ],
+            [("w", "with"), ("o", "orient"), ("B", "breaks")],
+        ),
+        #
+        # FOR SQLCMD
+        #
+        # for schema/s
+        (
+            SqlCmdMagic.execute,
+            "sqlcmd",
+            ["--schema", "--schema"],
+            [],
+        ),
+        (
+            SqlCmdMagic.execute,
+            "sqlcmd",
+            ["-s", "-s"],
+            [],
+        ),
+        (
+            SqlCmdMagic.execute,
+            "sqlcmd",
+            ["--schema", "-s"],
+            [("s", "schema")],
+        ),
+        # for table/t
+        (
+            SqlCmdMagic.execute,
+            "sqlcmd",
+            ["--table", "--table"],
+            [],
+        ),
+        (
+            SqlCmdMagic.execute,
+            "sqlcmd",
+            ["-t", "-t"],
+            [],
+        ),
+        (
+            SqlCmdMagic.execute,
+            "sqlcmd",
+            ["--table", "-t"],
+            [("t", "table")],
+        ),
+        # for mixed
+        (
+            SqlCmdMagic.execute,
+            "sqlcmd",
+            ["--table", "-t", "-s", "-s", "--schema"],
+            [("t", "table"), ("s", "schema")],
+        ),
+    ],
+)
+def test_check_duplicate_arguments_raises_usageerror(
+    check_duplicate_message_factory,
+    magic_execute,
+    cmd_from,
+    args,
+    aliases,
+):
+    with pytest.raises(UsageError) as excinfo:
+        util.check_duplicate_arguments(
+            magic_execute,
+            cmd_from,
+            args,
+            ALLOWED_DUPLICATES[cmd_from],
+            DISALLOWED_ALIASES[cmd_from],
+        )
+    assert check_duplicate_message_factory(cmd_from, args, aliases) in str(
+        excinfo.value
+    )
+
+
+@pytest.mark.parametrize(
+    "magic_execute, args, cmd_from",
+    [
+        (SqlMagic.execute, ["--creator"], "sql"),
+        (SqlMagic.execute, ["-c"], "sql"),
+        (SqlMagic.execute, ["--persist"], "sql"),
+        (SqlMagic.execute, ["-p"], "sql"),
+        (SqlMagic.execute, ["--persist", "--no-index"], "sql"),
+        (SqlMagic.execute, ["--persist", "-n"], "sql"),
+        (SqlMagic.execute, ["--file"], "sql"),
+        (SqlMagic.execute, ["-f"], "sql"),
+        (SqlMagic.execute, ["--save"], "sql"),
+        (SqlMagic.execute, ["-S"], "sql"),
+        (SqlMagic.execute, ["--alias"], "sql"),
+        (SqlMagic.execute, ["-A"], "sql"),
```

Comment:
`SqlMagic.execute`, `SqlPlotMagic.execute` etc are getting repeated in most cases. Instead this test can be broken down into 3 smaller tests each for `SqlMagic.execute`, `SqlPlotMagic.execute` and `SqlCmdMagic` and `magic_execute` parameter can be removed

Reply:
broke down tests into 3 [here](https://github.com/ploomber/jupysql/pull/908/commits/e68db1adcf0829128e99248223519e1faf8e696b)